libs/ui-lib/lib/ocm/components/AddHosts/day2Wizard/Day2ClusterDetails.tsx

```diff
@@ -104,6 +157,44 @@ const Day2ClusterDetails = () => {
     [day2Cluster.id, wizardContext],
   );
 
+  const changeCpuArchitectureDropdownAsync = React.useCallback(
+    async (value: string, initialValues: Day2ClusterDetailValues) => {
+      if (!pullSecret) return;
+      if (value !== initialValues.cpuArchitecture) {
+        //Fetch infraEnv by clusterId and cpu architecture
+        let infraEnv: InfraEnv | null;
+        try {
+          infraEnv = await InfraEnvsService.getInfraEnv(
+            day2Cluster.id,
+            value as SupportedCpuArchitecture,
+          );
+          if (!infraEnv) {
+            try {
+              infraEnv = await createNewInfraEnvWithCpuArchitecture(
+                pullSecret,
+                value as ClusterCpuArchitecture,
+                day2Cluster.id,
+                day2Cluster.name,
+                day2Cluster.openshiftVersion,
+              );
+            } catch (error) {
+              throw Error('Could not create infraEnv for this cpu architecture');
+            }
+          }
+        } catch (error) {
+          throw Error('Could not create infraEnv for this cpu architecture');
```

Comment:
Bad idea...

Reply:
I've added this, to show the "Error state":
if (!initialValues) {
    return <LoadingState />;
  } else if (initialValues instanceof Error) {
    return <ErrorState content="Failed to load associated data for the Day2 cluster" />;
  }